UI/src/app/services/backend-connector.service.ts

```diff
@@ -1,192 +1,203 @@
-    private _data: BehaviorSubject<Map<number, Trace[]>> = new BehaviorSubject<Map<number, Trace[]>>(new Map());
-    private _options: BehaviorSubject<Map<number, Options>> = new BehaviorSubject<Map<number, Options>>(new Map());
-    private _layout: BehaviorSubject<Map<number, Layout>> = new BehaviorSubject<Map<number, Layout>>(new Map());
-    public data: Observable<Map<number, Trace[]>> = this._data.asObservable();
-    public options: Observable<Map<number, Options>> = this._options.asObservable();
-    public layout: Observable<Map<number, Layout>> = this._layout.asObservable();
 
     constructor(private http: HttpClient) {
     }
 
+    private createPlotData(queryResponse: QueryResponse) {
+        if (queryResponse) {
+            const trace = new Trace(queryResponse.plot_type as PlotType);
+            trace.x = queryResponse.x;
+            trace.xaxis = queryResponse.x_title;
+            trace.y = queryResponse.y;
+            trace.yaxis = queryResponse.y_title;
+            trace.labels = queryResponse.x;
+            trace.values = queryResponse.y;
+
+            const layout = new Layout(queryResponse.title);
+
+            const options = new Options();
+            return new Data([trace], layout, options);
+
+        }
+
+    }
+
     public getData(userInput: string, datasetName: string) {
-        
-        return this.http
-            .get('http://localhost:3001/api/nlptodata', 
+
+        return this.http.get('/api/plot',
             {
                 params: {
                     userquery: userInput,
                     dataset: datasetName
                 }
-            })
-            .map((respose: Response) =>
-                respose.json() || {}
-            )
-            .subscribe((data) => {
-                console.log("Backendconnector" + data);
-                if(data) {
-                    const val = Data.parseData(data as JsonData);
-                    this.parseData(val.traces);
-                    this.parseLayout(val.layout);
-                    this.parseOptions(val.options);
-                }
-            });
-    }
+            }).map((queryResponse: QueryResponse) => {
+            return this.createPlotData(queryResponse);
 
-    private parseData(data: Trace[]){
-        const map = new Map<number, Trace[]>();
-        map.set(0, data);
-        this._data.next(map);
+        })
     }
 
-    private parseOptions(data: Options){
-        const map = new Map<number, Options>();
-        map.set(0, data);
-        this._options.next(map);
-    }
+    public update(userInput: string, data: Data) {
 
-    private parseLayout(data: Layout){
-        const map = new Map<number, Layout>();
-        map.set(0, data);
-        this._layout.next(map);
-    }
+        return this.http.post('https://js2018-group4.azurewebsites.net/api/nlp',
+            {
+                sentence: userInput,
+                data: data.traces,
+                layout: data.layout
```

Comment:
Did someone test this? I am pretty sure we need to manipulate the header here a bit.